### PR TITLE
feat(condo): DOMA-6090 no receipts for proceeding recurrent payment

### PR DESCRIPTION
**Pull request by ekabardinsky**

### CODE REVIEW

File: apps/condo/domains/acquiring/utils/taskSchema/index.js [hunk 1]
```diff
@@ -458,6 +460,80 @@ async function sendTomorrowPaymentNotificationSafely (context, recurrentPaymentC
     }
 }
 
+async function sendTomorrowPaymentNoReceiptsNotificationSafely (context, recurrentPaymentContext) {
+    if (isNil(recurrentPaymentContext) || isNil(get(recurrentPaymentContext, 'id'))) throw new Error('invalid recurrentPaymentContext argument')
```
Comment (Alllex202) [inline apps/condo/domains/acquiring/utils/taskSchema/index.js:464]: maybe add logger too

File: apps/condo/domains/acquiring/utils/taskSchema/index.js [hunk 2]
```diff
@@ -458,6 +460,80 @@ async function sendTomorrowPaymentNotificationSafely (context, recurrentPaymentC
...
+                },
+            },
+        })
+    } catch (error) {
+        logger.error({ msg: 'sendMessage error', error })
+    }
+}
+
+async function sendNoReceiptsToProceedNotificationSafely (context, recurrentPaymentContext) {
+    if (isNil(recurrentPaymentContext) || isNil(get(recurrentPaymentContext, 'id'))) throw new Error('invalid recurrentPaymentContext argument')
```
Comment (Alllex202) [inline apps/condo/domains/acquiring/utils/taskSchema/index.js:501]: and here

File: apps/condo/migrations/20230518201916-0273_alter_notificationusersetting_messagetype.js
Comment (Alllex202) [inline apps/condo/migrations/20230518201916-0273_alter_notificationusersetting_messagetype.js:1]: empty migration?

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/acquiring/utils/taskSchema/index.js (hunk 1 above)
Comment (ekabardinsky) [reply apps/condo/domains/acquiring/utils/taskSchema/index.js:464]: we have logger at the top level of the payment processing, all throwed errors will appears there

File: apps/condo/domains/acquiring/utils/taskSchema/index.js (hunk 2 above)
Comment (ekabardinsky) [reply apps/condo/domains/acquiring/utils/taskSchema/index.js:501]: same

File: apps/condo/domains/acquiring/tasks/notifyBeforeRecurrentPaymentDate.js [hunk 3]
```diff
@@ -10,11 +10,38 @@ const {
...
+    // filter receipts if they are already paid
+    billingReceipts = await filterPaidBillingReceipts(context, billingReceipts)
+
+    // send notification by count of payable receipts
+    if (billingReceipts.length === 0) {
+        return await sendTomorrowPaymentNoReceiptsNotificationSafely(context, recurrentPaymentContext)
+    } else {
+        return await sendTomorrowPaymentNotificationSafely(context, recurrentPaymentContext)
+    }
+}
```
[the remark this answers is not in the document]
Comment (ekabardinsky) [reply apps/condo/domains/acquiring/tasks/notifyBeforeRecurrentPaymentDate.js:43]: will change, 10x

**Pull request merged**